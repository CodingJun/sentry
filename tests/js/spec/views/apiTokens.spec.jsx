import PropTypes from 'prop-types';
import React from 'react';
import {shallow, mount} from 'enzyme';

import {Client} from 'app/api';
import ApiTokens from 'app/views/settings/account/apiTokens';

describe('ApiTokens', function() {
  beforeEach(function() {
    Client.clearMockResponses();
  });

  it('renders empty result', function() {
    Client.addMockResponse({
      url: '/api-tokens/',
    });

    let wrapper = shallow(<ApiTokens />, {
      childContextTypes: {
        router: PropTypes.object,
      },
      context: {
        router: TestStubs.router(),
      },
    });

    // Should be loading
    expect(wrapper).toMatchSnapshot();
  });

  it('renders with result', function() {
    Client.addMockResponse({
      url: '/api-tokens/',
      body: [TestStubs.ApiToken()],
    });

    let wrapper = mount(<ApiTokens />, {
      childContextTypes: {
        router: PropTypes.object,
      },
      context: {
        router: TestStubs.router(),
      },
    });

    // Should be loading
    expect(wrapper).toMatchSnapshot();
  });

  it('can delete token', function() {
    Client.addMockResponse({
      url: '/api-tokens/',
      body: [TestStubs.ApiToken()],
    });
    const resp = {
      url: '/api-tokens/',
      method: 'DELETE',
    };
    Client.addMockResponse(resp);

    expect(Client.getCallCount(resp)).toBe(0);

    let wrapper = mount(<ApiTokens />, {
      childContextTypes: {
        router: PropTypes.object,
      },
      context: {
        router: TestStubs.router(),
      },
    });

    wrapper.find('.icon-trash').simulate('click');

    // Should be loading
    expect(Client.getCallCount(resp)).toBe(1);
  });
});
