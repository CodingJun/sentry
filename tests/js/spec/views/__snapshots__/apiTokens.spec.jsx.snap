// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ApiTokens renders empty result 1`] = `
<DocumentTitle
  title="API Tokens - Sentry"
>
  <div>
    <SettingsPageHeading
      action={
        <Button
          className="ref-create-token"
          disabled={false}
          priority="primary"
          size="small"
          to="/settings/account/api/auth-tokens/new-token/"
        >
          Create New Token
        </Button>
      }
      label="Auth Tokens"
    />
    <p>
      Authentication tokens allow you to perform actions against the Sentry API on behalf of your account. They're the easiest way to get started using the API.
    </p>
    <p>
      <span
        key="5"
      >
        <span
          key="0"
        >
          For more information on how to use the web API, see our 
        </span>
        <a
          href="https://docs.sentry.io/hosted/api/"
          key="2"
        >
          <span
            key="1"
          >
            documentation
          </span>
        </a>
        <span
          key="3"
        >
          .
        </span>
      </span>
    </p>
    <p>
      <small>
        psst. Looking for the 
        <strong>
          DSN
        </strong>
         for an SDK? You'll find that under
         
        <strong>
          [Project] » Settings » Client Keys
        </strong>
        .
      </small>
    </p>
    <Styled(div)>
      <PanelHeader
        disablePadding={true}
      >
        <Flex
          align="center"
        >
          <Box
            flex="1"
            px={2}
          >
            Auth Token
          </Box>
        </Flex>
      </PanelHeader>
      <Styled(div)>
        <Styled(div)>
          You haven't created any authentication tokens yet.
        </Styled(div)>
      </Styled(div)>
    </Styled(div)>
  </div>
</DocumentTitle>
`;

exports[`ApiTokens renders with result 1`] = `
.glamor-6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: top;
  -webkit-box-align: top;
  -ms-flex-align: top;
  align-items: top;
  font-size: 14px;
  box-shadow: inset 0 -1px 0;
  margin-bottom: 30px;
}

.glamor-4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
}

.glamor-2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  font-weight: bold;
  border-bottom: 3px solid;
}

.glamor-0 {
  padding: 0 0 15px;
}

.glamor-42 {
  background: #fff;
  border: 1px solid;
  margin-bottom: 30px;
  position: relative;
}

.glamor-14 {
  border-bottom: 1px solid;
  border-radius: 0 0;
  background: padding:15px 0;
  text-transform: uppercase;
  font-size: 13px;
}

.glamor-12 {
  font-size: 12px;
  font-weight: 600;
  text-transform: uppercase;
  margin-bottom: 20px;
  font-size: inherit;
  text-transform: inherit;
  margin: 0;
}

.glamor-10 {
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.glamor-8 {
  box-sizing: border-box;
  padding-left: 16px;
  padding-right: 16px;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
}

.glamor-36 {
  box-sizing: border-box;
  padding-left: 16px;
  padding-right: 16px;
  padding-top: 16px;
  padding-bottom: 16px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  border-bottom: 1px solid;
}

.glamor-36:last-child {
  border: 0;
}

.glamor-30 {
  box-sizing: border-box;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
}

.glamor-22 {
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  overflow: hidden;
}

.glamor-18 {
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.glamor-26 {
  font-size: 0.9em;
}

.glamor-28 {
  font-size: 0.9em;
  color: #999;
}

.glamor-32 {
  box-sizing: border-box;
  padding-left: 16px;
}

<ApiTokens>
  <DocumentTitle
    title="API Tokens - Sentry"
  >
    <div>
      <SettingsPageHeading
        action={
          <Button
            className="ref-create-token"
            disabled={false}
            priority="primary"
            size="small"
            to="/settings/account/api/auth-tokens/new-token/"
          >
            Create New Token
          </Button>
        }
        label="Auth Tokens"
      >
        <Styled(div)>
          <div
            className="glamor-6 glamor-7"
          >
            <Styled(div)>
              <div
                className="glamor-4 glamor-5"
              >
                <Styled(div)>
                  <div
                    className="glamor-2 glamor-3"
                  >
                    <Styled(span)>
                      <span
                        className="glamor-0 glamor-1"
                      >
                        Auth Tokens
                      </span>
                    </Styled(span)>
                  </div>
                </Styled(div)>
              </div>
            </Styled(div)>
            <div>
              <Button
                className="ref-create-token"
                disabled={false}
                priority="primary"
                size="small"
                to="/settings/account/api/auth-tokens/new-token/"
              >
                <Link
                  className="ref-create-token button button-primary button-sm"
                  disabled={false}
                  onClick={[Function]}
                  onlyActiveOnIndex={false}
                  role="button"
                  style={Object {}}
                  to="/settings/account/api/auth-tokens/new-token/"
                >
                  <a
                    className="ref-create-token button button-primary button-sm"
                    disabled={false}
                    onClick={[Function]}
                    role="button"
                    style={Object {}}
                  >
                    <FlowLayout
                      truncate={false}
                    >
                      <div
                        className="flow-layout"
                      >
                        <span
                          className="button-label"
                        >
                          Create New Token
                        </span>
                      </div>
                    </FlowLayout>
                  </a>
                </Link>
              </Button>
            </div>
          </div>
        </Styled(div)>
      </SettingsPageHeading>
      <p>
        Authentication tokens allow you to perform actions against the Sentry API on behalf of your account. They're the easiest way to get started using the API.
      </p>
      <p>
        <span
          key="5"
        >
          <span
            key="0"
          >
            For more information on how to use the web API, see our 
          </span>
          <a
            href="https://docs.sentry.io/hosted/api/"
            key="2"
          >
            <span
              key="1"
            >
              documentation
            </span>
          </a>
          <span
            key="3"
          >
            .
          </span>
        </span>
      </p>
      <p>
        <small>
          psst. Looking for the 
          <strong>
            DSN
          </strong>
           for an SDK? You'll find that under
           
          <strong>
            [Project] » Settings » Client Keys
          </strong>
          .
        </small>
      </p>
      <Styled(div)>
        <div
          className="glamor-42 glamor-43"
        >
          <PanelHeader
            disablePadding={true}
          >
            <Styled(div)
              disablePadding={true}
            >
              <div
                className="glamor-14 glamor-15"
              >
                <Styled(div)>
                  <div
                    className="glamor-12 glamor-13"
                  >
                    <Flex
                      align="center"
                    >
                      <Base
                        align="center"
                        className="glamor-10"
                      >
                        <div
                          className="glamor-10"
                          is={null}
                        >
                          <Box
                            flex="1"
                            px={2}
                          >
                            <Base
                              className="glamor-8"
                              flex="1"
                              px={2}
                            >
                              <div
                                className="glamor-8"
                                is={null}
                              >
                                Auth Token
                              </div>
                            </Base>
                          </Box>
                        </div>
                      </Base>
                    </Flex>
                  </div>
                </Styled(div)>
              </div>
            </Styled(div)>
          </PanelHeader>
          <Styled(div)>
            <div
              className="glamor-40 glamor-41"
            >
              <ApiTokenRow
                key="apitoken123"
                onRemove={[Function]}
                token={
                  Object {
                    "dateCreated": 2017-10-17T11:41:20.000Z,
                    "scopes": Array [
                      "scope1",
                      "scope2",
                    ],
                    "token": "apitoken123",
                  }
                }
              >
                <Styled(Base)
                  justify="space-between"
                  px={2}
                  py={2}
                >
                  <Base
                    className="glamor-36 glamor-37"
                    justify="space-between"
                    px={2}
                    py={2}
                  >
                    <div
                      className="glamor-36 glamor-37"
                      is={null}
                    >
                      <Box
                        flex="1"
                      >
                        <Base
                          className="glamor-30"
                          flex="1"
                        >
                          <div
                            className="glamor-30"
                            is={null}
                          >
                            <div
                              style={
                                Object {
                                  "marginBottom": 5,
                                }
                              }
                            >
                              <small>
                                <TextCopyInput
                                  flexValueContainer={false}
                                  onCopy={[Function]}
                                  renderer={[Function]}
                                >
                                  <Styled(Base)>
                                    <Base
                                      className="glamor-22 glamor-23"
                                    >
                                      <div
                                        className="glamor-22 glamor-23"
                                        is={null}
                                      >
                                        <Styled(Base)
                                          flex=""
                                          onClick={[Function]}
                                        >
                                          <Base
                                            className="glamor-18 glamor-19"
                                            flex=""
                                            onClick={[Function]}
                                          >
                                            <div
                                              className="glamor-18 glamor-19"
                                              is={null}
                                              onClick={[Function]}
                                            >
                                              <Flex
                                                align="center"
                                              >
                                                <Base
                                                  align="center"
                                                  className="glamor-10"
                                                >
                                                  <div
                                                    className="glamor-10"
                                                    is={null}
                                                  >
                                                    apitoken123
                                                  </div>
                                                </Base>
                                              </Flex>
                                            </div>
                                          </Base>
                                        </Styled(Base)>
                                        <Clipboard
                                          errorMessage="Error copying to clipboard"
                                          hideMessages={false}
                                          hideUnsupported={true}
                                          successMessage="Copied to clipboard"
                                          value="apitoken123"
                                        />
                                      </div>
                                    </Base>
                                  </Styled(Base)>
                                </TextCopyInput>
                              </small>
                            </div>
                            <div
                              style={
                                Object {
                                  "marginBottom": 5,
                                }
                              }
                            >
                              <Styled(div)>
                                <div
                                  className="glamor-26 glamor-27"
                                >
                                  Created
                                   
                                  <DateTime
                                    date={2017-10-17T11:41:20.000Z}
                                    seconds={true}
                                  >
                                    <time>
                                      Oct 17, 2017 11:41:20 AM UTC
                                    </time>
                                  </DateTime>
                                </div>
                              </Styled(div)>
                            </div>
                            <div>
                              <Styled(div)>
                                <div
                                  className="glamor-28 glamor-29"
                                >
                                  scope1, scope2
                                </div>
                              </Styled(div)>
                            </div>
                          </div>
                        </Base>
                      </Box>
                      <Flex
                        align="center"
                      >
                        <Base
                          align="center"
                          className="glamor-10"
                        >
                          <div
                            className="glamor-10"
                            is={null}
                          >
                            <Box
                              pl={2}
                            >
                              <Base
                                className="glamor-32"
                                pl={2}
                              >
                                <div
                                  className="glamor-32"
                                  is={null}
                                >
                                  <Button
                                    disabled={false}
                                    onClick={[Function]}
                                  >
                                    <button
                                      className="button button-default"
                                      disabled={false}
                                      onClick={[Function]}
                                      role="button"
                                    >
                                      <FlowLayout
                                        truncate={false}
                                      >
                                        <div
                                          className="flow-layout"
                                        >
                                          <span
                                            className="button-label"
                                          >
                                            <span
                                              className="icon icon-trash"
                                            />
                                          </span>
                                        </div>
                                      </FlowLayout>
                                    </button>
                                  </Button>
                                </div>
                              </Base>
                            </Box>
                          </div>
                        </Base>
                      </Flex>
                    </div>
                  </Base>
                </Styled(Base)>
              </ApiTokenRow>
            </div>
          </Styled(div)>
        </div>
      </Styled(div)>
    </div>
  </DocumentTitle>
</ApiTokens>
`;
